### ISSUE TITLE
Using steward without the optional "tls" feature fails to compile

### ISSUE THREAD

**Issue by jvmncs** [shakacode/steward#4]
Hi, neat library!  Tried to pull this into a downstream project without the `tls` feature and found that it fails to compile that way (error below).

I think a common pattern to deal with this kind of issue would be to create a second implementation of `HttpService::https_connector` wrapped with `#[cfg(not(feature = "tls"))]` that just panics if you try to use it. Unfortunately, the `tls` dependency shows up in that method's signature, so this might require a small bit of refactoring (which is why I didn't just submit a PR to fix the issue).

```console
Checking steward v0.0.4
error[E0599]: no function or associated item named `https_connector` found for struct `HttpService` in the current scope
   --> /.../.cargo/registry/src/github.com-1ecc6299db9ec823/steward-0.0.4/src/net.rs:215:39
    |
116 | pub struct HttpService {
    | ---------------------- function or associated item `https_connector` not found for this struct
...
215 |                 let connector = Self::https_connector();
    |                                       ^^^^^^^^^^^^^^^
    |                                       |
    |                                       function or associated item not found in `HttpService`
    |                                       help: there is an associated function with a similar name: `http_connector`

error[E0599]: no function or associated item named `https_connector` found for struct `HttpService` in the current scope
   --> /.../.cargo/registry/src/github.com-1ecc6299db9ec823/steward-0.0.4/src/net.rs:236:39
    |
116 | pub struct HttpService {
    | ---------------------- function or associated item `https_connector` not found for this struct
...
236 |                 let connector = Self::https_connector();
    |                                       ^^^^^^^^^^^^^^^
    |                                       |
    |                                       function or associated item not found in `HttpService`
    |                                       help: there is an associated function with a similar name: `http_connector`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `steward` due to 2 previous errors
```

**Comment by alex35mil** [comment]
Hey @jvmncs, I'm totally fine with adding such feature. Mind to submit a PR?

**Comment by alex35mil** [comment]
Fixed in #5
